File: comparison/results/haiku/ponytail/humaneval/HumanEval_33/java/solution.java
public static ArrayList<Long> sortThird(ArrayList<Long> l) {
    List<Long> thirdIndices = new ArrayList<>();
    for (int i = 0; i < l.size(); i += 3) {
        thirdIndices.add(l.get(i));
    }
    
    Collections.sort(thirdIndices);
    
    ArrayList<Long> result = new ArrayList<>(l);
    int sortedIndex = 0;
    for (int i = 0; i < result.size(); i += 3) {
        result.set(i, thirdIndices.get(sortedIndex++));
    }
    
    return result;
}

File: comparison/results/haiku/ponytail/humaneval/HumanEval_33/typescript/solution.ts
function sort_third(l: number[]): number[] {
    const values_at_divisible_indices: number[] = [];
    
    for (let i = 0; i < l.length; i += 3) {
        values_at_divisible_indices.push(l[i]);
    }
    
    values_at_divisible_indices.sort((a, b) => a - b);
    
    const result = [...l];
    let sorted_index = 0;
    
    for (let i = 0; i < l.length; i += 3) {
        result[i] = values_at_divisible_indices[sorted_index];
        sorted_index++;
    }
    
    return result;
}
